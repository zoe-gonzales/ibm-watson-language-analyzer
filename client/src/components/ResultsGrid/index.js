import React from 'react';
import Title from '../Title';

const Grid = ({ type, text, results }) => {
    const generateRows = () => {
        if (type === "Emotion") {
            return (
                <div>
                    {/* User Input */}
                    <div className="row">
                        <div className="col s4">Text</div>
                        <div className="col s8">{text}</div>
                    </div>
                    {/* Headings */}
                    <div className="row">
                        <div className="col s4">{type}s</div>
                        <div className="col s4">{type}</div>
                        <div className="col s4">Score</div>
                    </div>
                    {/* Results */}
                    <div className="row">
                        <div className="col s4"></div>
                        <div className="col s4">Anger
                            <span role="img" aria-label="angry-face">
                            ðŸ˜ 
                            </span>
                        </div>
                        <div className="col s4">{results.Anger}</div>
                    </div>
                    <div className="row">
                        <div className="col s4"></div>
                        <div className="col s4">Disgust
                            <span role="img" aria-label="disgusted-face">
                            ðŸ˜–
                            </span>
                        </div>
                        <div className="col s4">{results.Disgust}</div>
                    </div>
                    <div className="row">
                        <div className="col s4"></div>
                        <div className="col s4">Fear
                            <span role="img" aria-label="fearful-face">
                            ðŸ˜¨
                            </span>
                        </div>
                        <div className="col s4">{results.Fear}</div>
                    </div>
                    <div className="row">
                        <div className="col s4"></div>
                        <div className="col s4">Joy
                            <span role="img" aria-label="joyful-face">
                            ðŸ˜„
                            </span>
                        </div>
                        <div className="col s4">{results.Joy}</div>
                    </div>
                    <div className="row">
                        <div className="col s4"></div>
                        <div className="col s4">Sadness
                            <span role="img" aria-label="sad-face">
                            ðŸ˜¢
                            </span>
                        </div>
                        <div className="col s4">{results.Sadness}</div>
                    </div>
                </div>
            )
        } else if (type === "Keyword") {
            return (
                <div>
                    {/* User Input */}
                    <div className="row">
                        <div className="col s3">Text</div>
                        <div className="col s9">{text}</div>
                    </div>
                    {/* Headings */}
                    <div className="row">
                        <div className="col s3">{type}s</div>
                        <div className="col s3">{type}</div>
                        <div className="col s3">Relevance</div>
                        <div className="col s3">Count</div>
                    </div>
                    {/* Results */}
                    {results.map(res => 
                        <div className="row" key={res.id}>
                            <div className="col s3"></div>
                            <div className="col s3">{res.Text}</div>
                            <div className="col s3">{res.Relevance}</div>
                            <div className="col s3">{res.Count}</div>
                        </div>
                    )}
                </div>
            )
        } else if (type === "Category") {
            return (
                <div>
                    {/* User Input */}
                    <div className="row">
                        <div className="col s4">Text</div>
                        <div className="col s8">{text}</div>
                    </div>
                    {/* Headings */}
                    <div className="row">
                        <div className="col s4">Categories</div>
                        <div className="col s4">{type}</div>
                        <div className="col s4">Score</div>
                    </div>
                    {/* Results */}
                    {results.map(res => 
                        <div className="row" key={res.id}>
                            <div className="col s4"></div>
                            <div className="col s4">{res.Label.substr(1)}</div>
                            <div className="col s4">{res.Score}</div>
                        </div>
                    )}
                </div>
            )
        }
    }

    return (
        <div>
            {/* Display results from API request; if not defined, display message */}
            <Title>Language Analyzer</Title>
            {results 
                ? generateRows()
                : (
                    <div className="row">
                        <div className="col s12">
                            Sorry, no results were found for your search. Please try again!
                        </div>
                    </div>
                )
            }
        </div>
    )
}

export default Grid;