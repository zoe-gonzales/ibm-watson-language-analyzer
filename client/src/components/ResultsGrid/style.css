.txt {
    font-size: 1.5rem;
}

.emoji {
    margin: 0rem .5rem;
}