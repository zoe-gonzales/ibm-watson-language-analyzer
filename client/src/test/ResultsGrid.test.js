import React from 'react';
import ResultsGrid from '../components/ResultsGrid';
import renderer from 'react-test-renderer';

it('renders ResultsGrid component correctly', () => {
    const input = 'Hello my name is Zoë'
    const typeOfResults = 'Keyword'
    const apiResults = [
        {
            keyword: 'Hello',
            relevance: 0.222
        },
        {
            keyword: 'name',
            relevance: 0.222
        },
        {
            keyword: 'Zoë',
            relevance: 0.222
        }
    ];
    const component = renderer
    .create(<ResultsGrid text={input} type={typeOfResults} results={apiResults} />)
    .toJSON();
    expect(component).toMatchSnapshot();
});