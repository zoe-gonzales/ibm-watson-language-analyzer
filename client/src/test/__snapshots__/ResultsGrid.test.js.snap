// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResultsGrid component correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s4"
    >
      Text
    </div>
    <div
      className="col s8"
    >
      Hello my name is Zoë
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col s4"
    >
      Keyword
      s
    </div>
    <div
      className="col s4"
    >
      Keyword
    </div>
    <div
      className="col s4"
    >
      Relevance
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col s4"
    />
    <div
      className="col s4"
    >
      Hello
    </div>
    <div
      className="col s4"
    >
      0.222
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col s4"
    />
    <div
      className="col s4"
    >
      name
    </div>
    <div
      className="col s4"
    >
      0.222
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col s4"
    />
    <div
      className="col s4"
    >
      Zoë
    </div>
    <div
      className="col s4"
    >
      0.222
    </div>
  </div>
</div>
`;
